src/hotspot/share/runtime/os.hpp

```diff
@@ -100,19 +100,30 @@ class os: AllStatic {
 #endif
 
  public:
-  enum { page_sizes_max = 9 }; // Size of _page_sizes array (8 plus a sentinel)
+
+  // A simple value class holding a set of page sizes (similar to sigset_t)
+  class PagesizeSet {
+    uintx _v;
```

Comment:
All in-parameters are size_t. It seems like this could be size_t as well.

Reply:
This is my personal taste; I dislike keeping anything a size_t which is not a memory range size (e.g. I dislike its use as an array index). But it does not matter much to me here since size_t is guaranteed to be unsigned.